src/rpc/net.cpp

```diff
@@ -446,7 +460,34 @@ UniValue getdeprecationinfo(const UniValue& params, bool fHelp)
     obj.pushKV("version", CLIENT_VERSION);
     obj.pushKV("subversion",
         FormatSubVersion(CLIENT_NAME, CLIENT_VERSION, std::vector<string>()));
-    obj.pushKV("deprecationheight", DEPRECATION_HEIGHT);
+
+    if (chainparams.NetworkIDString() == "main") {
+        if (fEnableDeprecationInfoDeprecationHeight) {
+            obj.pushKV("deprecationheight", DEPRECATION_HEIGHT);
+        }
+
+        UniValue eos(UniValue::VOBJ);
+        eos.pushKV("block_height", DEPRECATION_HEIGHT);
+        {
+            LOCK(cs_main);
+            auto blocksToDeprecation = DEPRECATION_HEIGHT - chainActive.Height();
+
+            eos.pushKV("estimated_time", GetTime() + (blocksToDeprecation * EXPECTED_BLOCKS_PER_HOUR * 60 * 60));
```

Comment:
This calculation has incorrect units:

$$
\begin{aligned}
\mathsf{curTime} &+ \left( \mathsf{blocksToDeprecation} * \mathsf{expectedBlocksPerHour} * 60 * 60 \right) \\
= \mathsf{seconds} &+ \left( \mathsf{blocks} * \frac{\mathsf{blocks}}{\mathsf{hours}} * \frac{\mathsf{minutes}}{\mathsf{hours}} * 
\frac{\mathsf{seconds}}{\mathsf{minutes}} \right) \\
= \mathsf{seconds} &+ \left( \frac{\mathsf{blocks}^2 \cdot \mathsf{seconds}}{\mathsf{hours}^2} \right) \\
\end{aligned}
$$

What we want is:

$$
\begin{aligned}
\mathsf{curTime} &+ \left( \frac{\mathsf{blocksToDeprecation}}{\mathsf{expectedBlocksPerHour}} * 60 * 60 \right) \\
= \mathsf{seconds} &+ \left( \mathsf{blocks} * \frac{\mathsf{hours}}{\mathsf{blocks}} * \frac{\mathsf{minutes}}{\mathsf{hours}} * 
\frac{\mathsf{seconds}}{\mathsf{minutes}} \right) \\
= \mathsf{seconds} & \\
\end{aligned}
$$

or more simply:

$$
\begin{aligned}
\mathsf{curTime} &+ \left( \mathsf{blocksToDeprecation} * \mathsf{expectedSecondsPerBlock} \right) \\
= \mathsf{seconds} &+ \left( \mathsf{blocks} * \frac{\mathsf{seconds}}{\mathsf{blocks}} \right) \\
= \mathsf{seconds} & \\
\end{aligned}
$$
```suggestion
            eos.pushKV("estimated_time", GetTime() + (blocksToDeprecation * Consensus::POST_BLOSSOM_POW_TARGET_SPACING));
```

We should also move this calculation into an `int64_t EstimatedTimeOfEoS(int nHeight)` function in `src/deprecation.cpp` (just above `EnforceNodeDeprecation`) (and then maybe factor out a common `static int BlocksToEoS(int nHeight)` local function in `src/deprecation.cpp` to ensure the calculation is consistent).